zig cc segfault when linking against macOS frameworks

Here's a simple test case:
```
int test(void) {
    return 1;
}
```

Running with `zig cc` and trying to link against `Cocoa` produces a segfault:
```
zig cc -F /System/Library/Frameworks -framework Cocoa darwin.c

fish: 'zig cc -F /Syste…' terminated by signal SIGSEGV (Address boundary error)
```

Running with system `cc` "works" (up to the point of printing an error):
```
cc -F /System/Library/Frameworks -framework Cocoa darwin.c

Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

If you remove `-framework Cocoa`, `zig cc` similarly "works":
```
Undefined symbol: C entry: _main
symbol(s) not found
error: LLDReportedFailure
```

Interestingly, if I add `-c` to the `zig cc` invocation, it works fine (producing an object file), so maybe this is just an issue in error handling (when `_main` is missing)?

--- a/src/link/MachO.zig
+++ b/src/link/MachO.zig
@@ -835,7 +835,10 @@ fn linkWithLLD(self: *MachO, comp: *Compilation) !void {
                             std.process.exit(1);
                         }
                     },
-                    else => std.process.abort(),
+                    else => {
+                        log.err("{s} terminated", .{ argv.items[0] });
+                        return error.LLDCrashed;
+                    },
                 }
             } else {
                 child.stdin_behavior = .Ignore;

